app/models/test_track/session.rb

```diff
@@ -11,34 +11,25 @@ def initialize(controller)
   def manage
     yield
   ensure
-    manage_cookies!
-    manage_response_headers!
+    if visitor.id_loaded?
+      manage_cookies!
+      manage_response_headers!
+    end
     notify_unsynced_assignments! if sync_assignments?
   end
 
   def visitor_dsl_for(identity)
-    if has_matching_identity?(identity)
-      visitor_dsl
-    else
-      TestTrack::VisitorDSL.new(visitors_by_identity[identity])
-    end
+    TestTrack::VisitorDSL.new(visitors_by_identity[identity])
```

Comment:
Now we're not special casing the identity-that-matches-the-controller-user - all visitor's being located by identity hit the same endpoint. This is the crux of the change - never trust a relationship between a visitor_id from the controller and an authenticated identity.